evaluator/autoware_control_evaluator/src/control_evaluator_node.cpp

```diff
@@ -33,12 +37,63 @@ ControlEvaluatorNode::ControlEvaluatorNode(const rclcpp::NodeOptions & node_opti
     this->create_publisher<tier4_debug_msgs::msg::Float64Stamped>("~/debug/processing_time_ms", 1);
   metrics_pub_ = create_publisher<MetricArrayMsg>("~/metrics", 1);
 
+  // Parameters
+  output_metrics_ = declare_parameter<bool>("output_metrics");
+
   // Timer callback to publish evaluator diagnostics
   using namespace std::literals::chrono_literals;
   timer_ =
     rclcpp::create_timer(this, get_clock(), 100ms, std::bind(&ControlEvaluatorNode::onTimer, this));
 }
 
+ControlEvaluatorNode::~ControlEvaluatorNode()
+{
+  if (!output_metrics_) {
+    return;
+  }
+
+  // generate json data
+  using json = nlohmann::json;
+  json j;
+  for (Metric metric : metrics_) {
+    const std::string base_name = metric_to_str.at(metric) + "/";
+    j[base_name + "min"] = metric_accumulators_[static_cast<size_t>(metric)].min();
+    j[base_name + "max"] = metric_accumulators_[static_cast<size_t>(metric)].max();
+    j[base_name + "mean"] = metric_accumulators_[static_cast<size_t>(metric)].mean();
+    j[base_name + "count"] = metric_accumulators_[static_cast<size_t>(metric)].count();
+    j[base_name + "description"] = metric_descriptions.at(metric);
+  }
```

Comment:
if writing like
```
for (size_t i = 0; i < static_cast<sizse_t>(Metric::SIZE); ++i) {
j[base_name + "min"] = metric_accumulators_[i].min();
```
metrics_ variable is not necessary?

Reply:
I want to get `metric_descriptions`:
`j[base_name + "description"] = metric_descriptions.at(metric);`